src/common/utils.hpp

```diff
@@ -538,7 +538,8 @@ const char *format_cvt_impl(T &&t) {
 
 template <typename... Args>
 std::string format_impl(const char *fmt, Args... args) {
-    size_t sz = snprintf(nullptr, 0, fmt, args...);
+    // volatile here is a workaround for GCC format-truncation warning
```

Comment:
In the comment could you specify the compiler version that this fixes as well as the warning itself?